tests/test_user_security.py

```diff
@@ -0,0 +1,349 @@
+
+import jwt
+from django.conf import settings
+from django.http import HttpRequest
+from freezegun import freeze_time
+from sqlalchemy import Integer, Column, VARCHAR
+
+from core_lib.cache.cache_decorator import Cache
+from core_lib.cache.cache_handler_ram import CacheHandlerRam
+from core_lib.core_lib import CoreLib
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+from core_lib.data_layers.data.db.sqlalchemy.types.int_enum import IntEnum
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.data_transform.result_to_dict import result_to_dict
+from core_lib.rule_validator.rule_validator import ValueRuleValidator, RuleValidator
+from core_lib.session.jwt_token_handler import JWTTokenHandler
+from core_lib.session.security_handler import SecurityHandler
+from core_lib.session.user_security import UserSecurity
+from core_lib.web_helpers.django.decorators import RequireLogin
+from core_lib.web_helpers.request_response_helpers import response_status
+from core_lib.web_helpers.web_helprs_utils import WebHelpersUtils
+from tests.test_data.test_utils import connect_to_mem_db
+
+
+# CRUD SETUP
+class User(Base):
+    __tablename__ = 'user_security'
+
+    class PolicyRoles(enum.Enum):
+        ADMIN = 1
+        DELETE = 2
+        CREATE = 3
+        UPDATE = 4
+        USER = 5
+
+    class Status(enum.Enum):
+        ACTIVE = 1
+        NOT_ACTIVE = 0
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    username = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    role = Column('role', IntEnum(PolicyRoles), default=PolicyRoles.USER)
+    status = Column('status', IntEnum(Status), default=Status.ACTIVE)
+
+
+class UserDataAccess(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('username', str),
+        ValueRuleValidator('email', str),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, User, connect_to_mem_db(), UserDataAccess.rules_validator)
+
+
+# CRUD INIT
+user_data_access = UserDataAccess()
+
+
+@Cache(key='user_data_{u_id}', expire=timedelta(seconds=30))
+def get_user(u_id):
+    return result_to_dict(user_data_access.get(u_id))
+
+
+# USER SECURITY SETUP
+class SessionUser(object):
+    def __init__(self, id: int, email: str):
+        self.id = id
+        self.email = email
+
+    def __dict__(self):
+        return {'id': self.id, 'email': self.email}
+
+
+def has_access(user, check_policies):
+    role = 5
+    user_role = user['role']
+
+    status = 1
+    user_status = user['status']
+    for policy in check_policies:
+        if type(policy) == User.PolicyRoles:
+            role = policy.value
+        if type(policy) == User.Status:
+            status = policy.value
+    if user_role <= role and user_status == status:

```

Comment:
will this work. instead of  `user_status == status` do `user_status <= status`?

Reply:
it's like either active or not active, so `==` will work better here, if the user is not active then he/she won't get entry unless we pass the NOT_ACTIVE policy in the rules

Reply:
updated the status. revisit!